modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,487 @@
+    """
+
+    partition_type = PandasOnRayDataframePartition
+    instance_type = ray.ObjectRef
+    axis = None
+
+    def __init__(self, list_of_blocks, get_ip=False, full_axis=True, call_queue=None):
+        if isinstance(list_of_blocks, PandasOnRayDataframePartition):
+            list_of_blocks = [list_of_blocks]
+        self.full_axis = full_axis
+        self.call_queue = call_queue or []
+        # In the simple case, none of the partitions that will compose this
+        # partition are themselves virtual partition. The partitions that will
+        # be combined are just the partitions as given to the constructor.
+        if not any(
+            isinstance(o, PandasOnRayDataframeVirtualPartition) for o in list_of_blocks
+        ):
+            self.list_of_partitions_to_combine = list_of_blocks
+            return
+        # Check that all axis are the same in `list_of_blocks`
+        assert (
+            len(
+                set(
+                    o.axis
+                    for o in list_of_blocks
+                    if isinstance(o, PandasOnRayDataframeVirtualPartition)
+                )
+            )
+            == 1
+        )
+        # When the axis of all virtual partitions matches this axis, extend and combine the lists of
+        # physical partitions
+        if (
+            next(
+                o
+                for o in list_of_blocks
+                if isinstance(o, PandasOnRayDataframeVirtualPartition)
+            ).axis
+            == self.axis
+        ):
+            new_list_of_blocks = []
+            for o in list_of_blocks:
+                new_list_of_blocks.extend(
+                    o.list_of_partitions_to_combine
+                ) if isinstance(
+                    o, PandasOnRayDataframeVirtualPartition
+                ) else new_list_of_blocks.append(
+                    o
+                )
+            self.list_of_partitions_to_combine = new_list_of_blocks
+        # Materialize partitions if the axis of this virtual does not match the virtual partitions
+        else:
+            self.list_of_partitions_to_combine = [
+                obj.force_materialization().list_of_partitions_to_combine[0]
+                if isinstance(obj, PandasOnRayDataframeVirtualPartition)
+                else obj
+                for obj in list_of_blocks
+            ]
+
+    @property
+    def list_of_blocks(self):
+        """
+        Get the list of physical partition objects that compose this partition.
+
+        Returns
+        -------
+        List
+            A list of ``ray.ObjectRef``.
+        """
+        # Defer draining call queue until we get the partitions
+        # TODO Look into draining call queue at the same time as the task
+        for partition in self.list_of_partitions_to_combine:
+            partition.drain_call_queue()
+        return [o.oid for o in self.list_of_partitions_to_combine]
```

Comment:
```suggestion
        result = [None] * len(self.list_of_partitions_to_combine)
        for idx, partition in enumerate(idx, self.list_of_partitions_to_combine):
            partition.drain_call_queue()
            result[idx] = partition.oid
        return result
```
Won't this be faster?